### DISCUSSION TITLE
Audio Tools- Getting Pico working with an i2s dac using Platform IO

### DISCUSSION THREAD

**Discussion by SnazzyBlanket** [pschatzmann/arduino-audio-tools#1415] [Ideas]
I am putting this here for others who want to try using the RP2040 with Platoform IO and Arduino-Audio tools. 

i was struggling with getting the Dac to work UNTIL i found this thread

https://github.com/earlephilhower/arduino-pico/issues/1656

after reading through the thread I was able to get sound working here is the working code for a sine generator

`
/**
 * @file streams-generator-analog.ino
 * @author Phil Schatzmann
 * @brief see https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-generator-analog/README.md 
 * @author Phil Schatzmann
 * @copyright GPLv3
 **/
 #include <Arduino.h>
#include <I2S.h>

#include "WiFi.h"

#define LED_PIN 25
#include "AudioTools.h"
#include "AudioLibs/AudioSTK.h"

#include "AudioLibs/AudioBoardStream.h"

AudioInfo info(44100, 1, 16);
int16_t sine_array[] = {0, 4560, 9031, 13327, 17363, 21062, 24350, 27165, 29450, 31163, 32269, 32747, 32587, 31793, 30381, 28377, 25820, 22761, 19259, 15383, 11206, 6812, 2285, -2285, -6812, -11206, -15383, -19259, -22761, -25820, -28377, -30381, -31793, -32587, -32747, -32269, -31163, -29450, -27165, -24350, -21062, -17363, -13327, -9031, -4560  };
GeneratorFromArray<int16_t> sineWave(sine_array,0,false);
GeneratedSoundStream<int16_t> sound(sineWave);             // Stream generated from sine wave
I2SStream out;
StreamCopy copier(out, sound);                             // copies sound into i2s

// Arduino Setup
void setup(void) {  
  // Open Serial 
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Error);

  // start I2S
  I2SConfig i2sconfig = out.defaultConfig(TX_MODE);
 i2sconfig.buffer_count = 5;
  i2sconfig.buffer_size = 512;
  i2sconfig.pin_bck = 27;
  i2sconfig.pin_data = 26;
  i2sconfig.pin_ws = 28;
  out.begin(i2sconfig);

  // Setup sine wave
  sineWave.begin(info);
  Serial.println("started...");
}

// Arduino loop - copy sound to out 
void loop() {
  copier.copy();
}

`

**Comment by SnazzyBlanket** [comment]
and here is the exact board info file i used

_
[env:rpipico]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board = rpipico
framework = arduino
board_build.core = earlephilhower
board_build.filesystem_size = 0.5m
upload_protocol = cmsis-dap
lib_deps = 
	khoih-prog/RP2040_PWM@^1.7.0
	bodmer/TFT_eSPI@^2.5.43
platform_packages =
    maxgerhardt/framework-arduinopico@https://github.com/earlephilhower/arduino-pico.git_

**Comment by SnazzyBlanket** [comment]
this was with the PCM5102 Dac module

**Comment by pschatzmann** [comment]
Shouldn't you just better follow the [official instructions](https://arduino-pico.readthedocs.io/en/latest/platformio.html) ?

**Comment by SnazzyBlanket** [comment]
I am not going to worry about it. I spent 8 hours trying to get it to work and only found relief once i read that github thread. the good news it now works very well.

**Comment by SnazzyBlanket** [comment]
please keep in mind the way they had it in the offical instructions DID NOT WORK for me. it was only once i changed it to the way they had it in the github thread that it worked. 

however (and i can test this tomorow) IT Could also be because I added the logger line with ERROR instead of warning (which is what the original poster was talking about)

sorry by this point I am a bit exhausted

**Reply by pschatzmann** [reply]
No worries:. I just wanted the add the link to the official documentation, since this might change in the future.
And here is the link to [my instructions for PlatformIO ](https://github.com/pschatzmann/arduino-audio-tools/wiki/Working-with-PlatformIO)

**Comment by SnazzyBlanket** [comment]
thanks so much
